iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1772,6 +1772,10 @@ private List<String> getPartitions(DDLOperationContext context, Configuration jo
   public void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable, Map<String, String> partitionSpec)
       throws SemanticException {
     Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
+    if (hmsTable.getSnapshotRef() != null && hasUndergonePartitionEvolution(table)) {
```

Comment:
would this even be called if canUseTruncate returns false?

Reply:
it will be called before can use truncate, it is called as part of ``addTruncateTableOutputs``